File: app/schemas/neo4j.py
"""
Neo4j 相关的数据模式
"""
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any


class QueryRequest(BaseModel):
    """Cypher 查询请求"""
    query: str = Field(..., description="Cypher 查询语句（只读）")
    parameters: Optional[Dict[str, Any]] = Field(None, description="查询参数")


class SearchRequest(BaseModel):
    """节点搜索请求"""
    label: Optional[str] = Field(None, description="节点标签")
    property_key: Optional[str] = Field(None, description="属性键")
    property_value: Optional[str] = Field(None, description="属性值")
    limit: int = Field(50, ge=1, le=1000, description="返回数量限制")

File: frontend/src/services/neo4j.ts
import api from './api'

export interface Neo4jNode {
  id: number
  labels: string[]
  properties: Record<string, any>
}

export interface Neo4jRelationship {
  source: Neo4jNode
  relationship: {
    type: string
    properties: Record<string, any>
  }
  target: Neo4jNode
}

export interface QueryRequest {
  query: string
  parameters?: Record<string, any>
}

export interface SearchRequest {
  label?: string
  property_key?: string
  property_value?: string
  limit?: number
}

/** api 拦截器已解包为 response.data，此处显式标注返回类型 */
export interface Neo4jStats {
  nodes: Record<string, number>
  relationships: Record<string, number>
}

export interface QueryExecuteResult {
  results?: any[]
}

export interface NodesListResult {
  nodes?: Neo4jNode[]
}

export interface BrowserUrlResult {
  url: string
}

export const neo4jService = {
  async getStats(): Promise<Neo4jStats> {
    return api.get('/neo4j/stats') as Promise<Neo4jStats>
  },

  async executeQuery(request: QueryRequest): Promise<QueryExecuteResult> {
    return api.post('/neo4j/query', request) as Promise<QueryExecuteResult>
  },

  async searchNodes(request: SearchRequest): Promise<NodesListResult> {
    return api.post('/neo4j/search', request) as Promise<NodesListResult>
  },

  async getNodes(limit = 100): Promise<NodesListResult> {
    return api.get('/neo4j/nodes', { params: { limit } }) as Promise<NodesListResult>
  },

  async getNodeById(nodeId: number): Promise<any> {
    return api.get(`/neo4j/nodes/${nodeId}`)
  },

  async getRelationships(limit = 100) {
    return api.get('/neo4j/relationships', { params: { limit } })
  },

  async getBrowserUrl(): Promise<BrowserUrlResult> {
    return api.get('/neo4j/browser/url') as Promise<BrowserUrlResult>
  },
}
